openshift/privatelink-infrastructure-monitor
# First throw at structure and Feature

Rough first outline of how the quota scraping could be implemented.
It currently supports two different quotas:
1. Transitgateways per account and 
2. resource records per hosted zone
each support both getting the limit and current values.

## How to Test
To test run locally, do the following:
1. Log in to any aws account
2. build the app with 
```
go build
```
3. Then, run the resulting binary and pass an id to a hosted zone in your aws account:
```
./privatelink-infrastructure-monitor -hosted-zone-id <id_of_a_hosted_zone>
```

If all goes well, it should print something like this:
```
$ ./privatelink-infrastructure-monitor -hosted-zone-id <redacted>
transit_gateways_per_account
        Quota: 5.00
        Usage: 0.00
resource_records_per_hosted_zone_Z085168021NCJT8IKW4DC
        Quota: 10000.00
        Usage: 1384.00
```
